### DISCUSSION TITLE
Triggers for GPU & Power Modes (application trigger, charging trigger)

### DISCUSSION THREAD

**Discussion by t-AIR-e** [seerge/g-helper#1679] [Ideas]
First off, don't want to sound unappreciative, so first off, thanks Seerge! Love G-Helper! 

I have a couple of ideas that would make G-Helper 110%, at least for me!

USB-C trigger- At the moment, when I connect my laptop to a dock using USB-C, I currently have to manually switch the GPU from Optimized to Standard mode. I'd love to be able to manage GPU/power modes on USB-C trigger or dock triggers or if charging triggers.

For applications- When playing games that need the extra power, like 2077, it would awesome if I had a profile set-up to switch up the power mode from silent to balanced or turbo. GPU modes automatically switch so no need for changes there when set to optimized.

**Comment by seerge** [comment]
@TEA-AIR-E hello,

1. If you talk about dock that is also a charger then Optimized mode will turn on dGPU when plugged in, and turn off when unplugged (as this is the only purpose of Optimized) :) (unless you have ``Keep dGPU disabled`` on USB-C toggled in ``Extra``)

2. As for profiles and  detecting currently running apps, it's difficult (i was answering [same question recently](https://github.com/seerge/g-helper/issues/1080)). It's only 3 profiles by default and you have ``Fn+F5 / Fn+Shift+F5`` to switch between them forwads / backwards :)

**Comment by t-AIR-e** [comment]
> @TEA-AIR-E hello,
> 
> 1. If you talk about dock that is also a charger then Optimized mode will turn on dGPU when plugged in, and turn off when unplugged (as this is the only purpose of Optimized) :) (unless you have `Keep dGPU disabled` on USB-C toggled in `Extra`)
> 2. As for profiles and  detecting currently running apps, it's difficult (i was answering [same question recently](https://github.com/seerge/g-helper/issues/1080)). It's only 3 profiles by default and you have `Fn+F5 / Fn+Shift+F5` to switch between them forwads / backwards :)


Thanks! I just disablsed that setting, `keep dGPU disabled`. 🍻

As for the profiles - Yes, like Armory Crate where on app.exe launch switch to custom profile 2 for CyberPunk 2077.
I didn't know about the KB shortcuts- will definitely use these now!
